samples/extensions/calibrated_timestamps/calibrated_timestamps.h

```diff
@@ -0,0 +1,190 @@
+/* Copyright (c) 2023, Holochip Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Demonstrates and showcase calibrated timestamps extension related functionalities
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+static std::string time_domain_to_string(VkTimeDomainEXT input_time_domain);        // this returns a string the input time domain Enum value
+
+class CalibratedTimestamps : public ApiVulkanSample
+{
+  private:
+	bool                         is_time_domain_updated = false;        // this is just to tell if time domain update has a VK_SUCCESS in the end
+	bool                         is_timestamp_updated   = false;        // this is just to tell the GUI whether timestamps operation has a VK_SUCCESS in the end
+	std::vector<VkTimeDomainEXT> time_domains{};                        // this holds all time domains extracted from the current Instance
+	std::vector<uint64_t>        timestamps{};                          // timestamps vector
+	std::vector<uint64_t>        max_deviations{};                      // max deviations vector
+
+	struct
+	{
+		int             index = 0;
+		VkTimeDomainEXT timeDomainEXT{};
+	} selected_time_domain{};
+
+	struct DeltaTimestamp
+	{
+		uint64_t    begin = 0;
+		uint64_t    end   = 0;
+		uint64_t    delta = 0;
+		std::string tag   = "Untagged";
+
+		void get_delta()
```

Comment:
Is this really worth a function?
If at all, it doesn't get you anything (returning void) but calculates something. So `calculate_delta` or `determine_delta` would be a better fit.